Bottom: bcdcf7f7bde15e2c6e6ea0ddbe0cf8fc11348fdf
Top:    74fc20de38d12ad9d5dad8f454b175c0336872d6
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-16 15:26:11 +0800

Refresh of autofs-5.0.7-only-prove-specific-nfs-version-if-requested.patch

---

diff --git a/modules/mount_nfs.c b/modules/mount_nfs.c
index 82f674d..5a582ef 100644
--- a/modules/mount_nfs.c
+++ b/modules/mount_nfs.c
@@ -148,12 +148,12 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name, int
 					vers = NFS4_VERS_MASK | TCP_SUPPORTED;
 				else if (strncmp("vers=3", cp, o_len) == 0 ||
 					 strncmp("nfsvers=3", cp, o_len) == 0) {
-					vers &= ~NFS4_VERS_MASK;
-					vers |= (NFS3_REQUESTED & ~NFS_VERS_MASK);
+					vers &= ~(NFS4_VERS_MASK | NFS_VERS_MASK);
+					vers |= NFS3_REQUESTED;
 				} else if (strncmp("vers=2", cp, o_len) == 0 ||
 					 strncmp("nfsvers=2", cp, o_len) == 0) {
-					vers &= ~NFS4_VERS_MASK;
-					vers |= (NFS2_REQUESTED & ~NFS_VERS_MASK);
+					vers &= ~(NFS4_VERS_MASK | NFS_VERS_MASK);
+					vers |= NFS2_REQUESTED;
 				} else if (strstr(cp, "port=") == cp &&
 					 o_len - 5 < 25) {
 					char optport[25];
